internal/provider/resource_secret.go

```diff
@@ -0,0 +1,295 @@
+	Name types.String `tfsdk:"name"`
+	// Value of the secret to be added or updated. This attribute is required for 'add' and 'update' actions.
+	// Template: [<key>[#base64]]=<value>[ ...]
+	Value types.Map `tfsdk:"value"`
+	// SecretId is the ID of the secret to be updated or removed. This attribute is required for 'update' and 'remove' actions.
+	SecretId types.String `tfsdk:"secret_id"`
+	// Info is the description of the secret. This attribute is optional for all actions.
+	Info types.String `tfsdk:"info"`
+}
+
+func (s *secretResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+func (s *secretResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_secret"
+}
+
+func (s *secretResource) Schema(_ context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "A resource that represents a Juju secret.",
+		Attributes: map[string]schema.Attribute{
+			"model": schema.StringAttribute{
+				Description: "The model in which the secret belongs.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the secret.",
+				Optional:    true,
+			},
+			"value": schema.MapAttribute{
+				Description: "The value map of the secret. There can be more than one key-value pair.",
+				ElementType: types.StringType,
+				Required:    true,
+				Sensitive:   true,
+			},
+			"secret_id": schema.StringAttribute{
+				Description: "The ID of the secret.",
+				Computed:    true,
+			},
+			"info": schema.StringAttribute{
+				Description: "The description of the secret.",
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func (s *secretResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*juju.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *juju.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+	s.client = client
+	// Create the local logging subsystem here, using the TF context when creating it.
+	s.subCtx = tflog.NewSubsystem(ctx, LogResourceSecret)
+}
+
+func (s *secretResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Prevent panic if the provider has not been configured.
+	if s.client == nil {
+		addClientNotConfiguredError(&resp.Diagnostics, "secret", "create")
+		return
+	}
+
+	var plan secretResourceModel
```

Comment:
Should this var be called `plan`? I may be misunderstanding but it seems to be just the model the secret belongs to, not the whole plan.